File: app/src/main/java/live/bolder/hustest/MovieInterface.java
package live.bolder.hustest;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface MovieInterface {
    @GET( "/3/movie/{category}")
    Call<MovieResults> getMovies(
            @Path( "category" ) String category,
            @Query( "api_key" ) String apiKey,
            @Query( "language" ) String language,
            @Query( "page" ) int page
    );

    @GET( "/3/movie/{movie_id}")
    Call<MovieDetails> getMovieDetails(
            @Path( "movie_id" ) int movie_id,
            @Query( "api_key" ) String apiKey,
            @Query( "language" ) String language
    );
}

File: app/src/main/java/live/bolder/hustest/kotlin/MovieInterface.kt
package live.bolder.hustest.kotlin

import live.bolder.hustest.MovieDetails
import live.bolder.hustest.MovieResults
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface MovieInterface {

    @GET("/3/movie/{category}")
    suspend fun getMovies(
        @Path("category") category: String,
        @Query("api_key") apiKey: String,
        @Query("language") language: String,
        @Query("page") page: Int
    ): Response<MovieResults>

    @GET("/3/movie/{movie_id}")
    suspend fun getMovieDetails(
        @Path("movie_id") movie_id: Int,
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Response<MovieDetails>
}